web/assets/js/app.js

```diff
@@ -15,4 +15,35 @@ $(document).ready(function(){
         $('.navbar-header').addClass('navbar-show');
         $('#navbar').addClass('navbar-show').addClass('navbar-list-show');
     });
+
+    $("button.load-more").on('click', function() {
+       var url = $(this).data('url');
+       var page = $(this).data('page');
+       var $last = $(this).parent().parent().prev().children().last();
+       var $button = $(this);
+       var $parent = $(this).parent();
+
+       $(this).hide();
+       $parent.append('<div class="button-load-progress"></div>').fadeIn();
+
+        $.ajax({
+            type: 'POST',
+            url: url + '/' + page,
+            dataType: 'json',
+            success: function(data) {
+                if (false === data.showLoadMoreButton) {
+                    $button.parent().empty().hide();
+                }
+                $button.data('page', page + 1);
+                $(data.html).hide().insertAfter($last).fadeIn('slow');
+                $parent.find('div').remove();
```

Comment:
@kmadejski Similar note can be applied to `$last` variable. We could introduce something like `data-target` attribute with selector of an element where new pages will be appended, to avoid dependency on a concrete page structure.